src/Microsoft.Health.Fhir.Shared.Core/Features/Definition/SearchParameterDefinitionBuilder.cs

```diff
@@ -292,21 +281,21 @@ void EnsureNoIssues()
 
         private SearchParameterInfo CreateSearchParameterInfo(SearchParameter searchParameter)
         {
-            SearchParameterInfo info = searchParameter.ToInfo();
-
-            var uri = new Uri(searchParameter.Url);
-            if (_unsupportedParams.Unsupported.Contains(uri))
+            // Return SearchParameterInfo that has already been created for this Uri
+            if (_uriDictionary.TryGetValue(new Uri(searchParameter.Url), out var spi))
             {
-                info.IsSupported = false;
-                info.IsSearchable = false;
+                return spi;
             }
 
-            if (_unsupportedParams.PartialSupport.Contains(uri))
+            // Return SearchParameterInfo that has already been created for this Resource
+            if (_resourceTypeDictionary.TryGetValue(searchParameter.ResourceType.ToString(), out var spDictionary) &&
+                spDictionary.TryGetValue(searchParameter.Name, out var resourceSpi) &&
+                resourceSpi.Url == new Uri(searchParameter.Url))
```

Comment:
Curious, if we should compare Uri instances using equals() method?
https://docs.microsoft.com/en-us/dotnet/api/system.uri.equals?view=netframework-4.8

Reply:
Apparently in this case the same (https://docs.microsoft.com/en-us/dotnet/api/system.uri.op_equality?view=netcore-3.1)

```
This overload uses the Equals method to determine whether the two Uri instances are equivalent.
```